java-frontend/src/test/java/org/sonar/java/model/LiteralUtilsTest.java

```diff
@@ -179,6 +179,26 @@ public void is_0xff() {
     assertThat(LiteralUtils.is0xff(tree)).isFalse();
   }
 
+  public void hasBooleanValue_withNonBooleanLiteral_returnsFalse() {
+    ExpressionTree tree = getFirstExpression("void foo(java.util.Properties props){ props.setProperty(\"myKey\", \"myValue\"); }");
+    assertThat(LiteralUtils.hasValue(tree, false)).isFalse();
+    assertThat(LiteralUtils.hasValue(tree, true)).isFalse();
+  }
+
+  @Test
+  public void hasBooleanValue_withOpposedValue_returnsFalse() {
+    LiteralTree falseTree = (LiteralTree) getReturnExpression("void foo(java.util.Properties props){ return false; }");
```

Comment:
Do we really need "java.util.Properties props"?

Reply:
copy-paste artifact :)